src/devices/tpms_eezrv.c

```diff
@@ -55,36 +55,50 @@ static int tpms_eezrv_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     bitbuffer_invert(bitbuffer);
     pos = bitbuffer_search(bitbuffer, 0, pos, preamble_pattern, sizeof(preamble_pattern) * 8);
     if (pos >= bitbuffer->bits_per_row[0]) {
-        decoder_log(decoder, 2, __func__, "Preamble not found");
+        decoder_log(decoder, 3, __func__, "Preamble not found");
         return DECODE_ABORT_EARLY;
     }
-    if (pos + 7 * 8 > bitbuffer->bits_per_row[0]) {
+    if (pos + 8 * 8 > bitbuffer->bits_per_row[0]) {
         decoder_log(decoder, 2, __func__, "Length check fail");
         return DECODE_ABORT_LENGTH;
     }
-    uint8_t b[6]  = {0};
+    uint8_t b[7]  = {0};
     uint8_t cc[1] = {0};
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 16, cc, sizeof(cc) * 8);
     bitbuffer_extract_bytes(bitbuffer, 0, pos + 24, b, sizeof(b) * 8);
+
     // verify checksum
-    if ((add_bytes(b, 6) & 0xff) != cc[0]) {
+    int computed_checksum = add_bytes(b, sizeof(b));
+    if (computed_checksum > 0xff) 
+        computed_checksum |= 0x80;
+    
+    if ((computed_checksum & 0xff) != cc[0]) {
         decoder_log(decoder, 2, __func__, "Checksum fail");
         return DECODE_FAIL_MIC;
     }
     char id_str[7];
     sprintf(id_str, "%02x%02x%02x", b[0], b[1], b[2]);
     float pressure_kPa = b[3] * 2.5;
     int temperature_C  = b[4] - 50;
-    int flags          = b[5];
-    char flags_str[3];
-    sprintf(flags_str, "%x", flags);
+    int flags1          = b[5];
+    int flags2          = b[6];
+    int fast_leak       = 0;
+
+    fast_leak = (flags1 & 0x10) && !(flags1 & 0x20);   // Check for fast leak flag set and resolved flag NOT set
```

Comment:
perhaps write as
```
int leak_flag = (flags1 & 0x10);
int resolved_flag = (flags1 & 0x20);
int fast_leak = leak_flag && !resolved_flag;
```

Reply:
Code updated to note value of fast_leak_detected and fast_leak_resolved and to then determine reported fast_leak output field base on these.